File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(def drops {3 "Pling" 5 "Plang" 7 "Plong"})

(defn convert
  "Converts a number to its corresponding string of raindrop sounds."
  [num]
  (or 
   (not-empty (apply str (for [[div drop] drops :when (zero? (mod num div))] drop))) 
   (str num)))

File: python/raindrops/raindrops.py
"""Function to output a Raindrops string"""

DROPS = {3: 'Pling', 5: 'Plang', 7: 'Plong'}

def convert(number):
    result = ''.join(drop for divisor, drop in DROPS.items() if number % divisor == 0)
    return result or str(number)
